### ISSUE TITLE
[Accessibility support] va-prototype kit

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#6072]
### Description

[Review Joe's PR](https://github.com/department-of-veterans-affairs/va-prototype-kit/pull/96)

A number of accessibility related issues were raised for @humancompanion-usds and I'm looking into them. This is a ticket just to track the work.

### Issues
1. `va-icon` not having srtext - This seems like a bug between the prototype kit and the MCP. One of them should have caught that there was semantic meaning for this icon and added it. Also, the a11y tests should have caught it. 
2. What's a security code (CVV/CVC)? - I get this is a button and not a link but I would have hoped ADE would push folks towards va-details rather than a raw HTML button. Agree? I don't want to argue with them but I'm tempted to leave a comment on the PR.
3. `<hr class="vads-u-border-top--0 vads-u-margin-y--4" aria-hidden="true" />` - Do we need to hide horizontal rules!? TIL is that is true. We have a Divider component but I stopped short of putting just a regular, straightforward rule in there because it didn't seem necessary. But perhaps I was wrong.
4. [prefersReducedMotion](https://github.com/department-of-veterans-affairs/va-prototype-kit/pull/96/changes#diff-965129608fb2645bf51fd36f59515f3debd81801be3fa07c20fdc37f6d244a46R167) - We don't have anything, to my knowledge, in VADS about respecting this preferences, which we should. I'm not sure VADS has anything about motion. It's almost an anti-pattern to use motion at all, but clearly with our rotation of certain drawer handles we have some in VADS. Worth adding a section on motion perhaps in the /foundation section?
5. `<article role="none">` - Agree with this decision? I can add it to the templates if so.
6. [Address block changes](https://github.com/department-of-veterans-affairs/va-prototype-kit/pull/96/changes#diff-b693edd8b37d3393465162466b9f91126f1e8fe5cb3c32e57c0d3b08f94ec21fL68) - This is another one where I'm thinking "Has this bug been in this 'component' (it's just some CSS) forever and we just never noticed it!?" This, like Divider, feels like one where maybe we do need to convert it to a web-component or just update the markup?
7. [Focus state management](https://github.com/department-of-veterans-affairs/va-prototype-kit/pull/96/changes#diff-7c1f4b9ae6ae29f69586c4af43d5f51a8406bfb5dea7519ae1c60ae26ec90a7bR79) - Here Joe is changing this across all prototypes. Necessary? If so, I suspect this might be just the tip of the iceberg. Given all of the focus state management work we did, can I get some input from you and @amyleadem  on how I might get that into the prototype kit? Eek. This is worrisome.
